src/functions/assertions/HaveParameter.ps1

```diff
@@ -235,7 +206,12 @@
 
         if ($PSBoundParameters.Keys -contains "DefaultValue") {
             $parameterMetadata = Get-ParameterInfo $ActualValue | & $SafeCommands['Where-Object'] { $_.Name -eq $ParameterName }
-            $actualDefault = if ($parameterMetadata.DefaultValue) { $parameterMetadata.DefaultValue } else { "" }
+            $actualDefault = if ($parameterMetadata.DefaultValue) {
+                $parameterMetadata.DefaultValue
```

Comment:
Should we write some type of warning (@nohwnd ?) if `Get-ParameterInfo` doesn't return anything? Like `-DefaultValue is only supported for functions and scripts` and ignore the rest of this block? Only confusing to get a `but the default value was <empty>` when we actually don't know.

Reply:
Please, let me know if you're okay with the updates I made to tst\functions\assertions\HaveParameter.Tests.ps1.  Before I added the Ast logic to Get-ParameterInfo, the opening parenthesis for the param block had to be on the same line and the datatype had to be on the same line as the parameter name.  If you could also let me know how you'd like me to update the return value, when $parameterMetadata.DefaultValue is $null, I'd appreciate it.

Reply:
> Please, let me know if you're okay with the updates I made to tst\functions\assertions\HaveParameter.Tests.ps1. Before I added the Ast logic to Get-ParameterInfo, the opening parenthesis for the param block had to be on the same line and the datatype had to be on the same line as the parameter name.

Please revert and add a specific `It`-test for this param-scenario with a minimal function included in the test. The current change would likely be lost in a future format-commit. Ex.

```ps1
    It 'works without paramblock with parantheses on different line' {
        function adv {
            param
            (
                $param1,
                $param2 = '123'
            )
        }

        Get-Command adv | Should -HaveParameter 'param2' -DefaultValue '123'
    }
```

> If you could also let me know how you'd like me to update the return value, when $parameterMetadata.DefaultValue is $null, I'd appreciate it.

If `$parameterMetadata` exists but `DefaultValue` is null then the parameter exists, but does not have a default value (or it's null). In that scenario it should behave like now. 

I was thinking about a warning when `Get-ParameterInfo $ActualValue` doesn't return anything, prior to filtering by parameter-name. 🙂 I suggest we fix that later in a different PR when this is merged. Issue is tracked in #2286.

Reply:
I've added a dedicated test for a paramblock with the opening parenthesis on a new line.